src/modules/schema.js

```diff
@@ -154,11 +121,42 @@ export default class Schema {
   static async getSchema(id, dockApi) {
     // TODO: requires blob module
     if (id === 'invalid-format') {
-      throw new Error(`Incorrect schema format`, dockApi);
+      throw new Error('Incorrect schema format', dockApi);
     } else if (id === 'invalid-id') {
       throw new Error(`Invalid schema id ${id}`, dockApi);
     }
 
     return {};
   }
+
+  /**
+   * Gets the JSON schema spec from given JSON. Will either return the stored JSON schema or get
+   * it using HTTP or will throw error if cannot get.
+   * @param {object} json
+   * @returns {Promise<object>}
+   */
+  static async getJSONSchemaSpec(json) {
+    const schemaKey = '$schema';
+    if (json[schemaKey]) {
```

Comment:
use json.$schema

Reply:
ideally like `const schema = json.$schema;` then replace all `json[schemaKey]` references below with `schema`

Reply:
I need `schemaKey` for error but have cached the result of `json[schemaKey]` in a variable